test/stdlib/StringTraps.swift

```diff
@@ -104,7 +104,36 @@ StringTraps.test("UTF16ViewSubscript/endIndex")
   i = s.utf16.index(after: i)
   i = s.utf16.index(after: i)
   expectCrashLater()
-  s.utf16[i]
+  _ = s.utf16[i]
+}
+
+StringTraps.test("UTF16ViewIndex/offsetLimited")
+  .code {
+  let sa = "foo"
+  let u16a = sa.utf16
+  let s16 = sa + "🤦🏻‍♀️"
+  let u16 = s16.utf16
+
+  let iaBegin = u16a.index(sa.startIndex, offsetBy: 99, limitedBy: sa.endIndex)
+  expectNil(iaBegin)
+  let iaEnd = u16a.index(sa.endIndex, offsetBy: 99, limitedBy: sa.endIndex)
```

Comment:
It would also be interesting to test for non-positive offsets starting from an `endIndex`.

Reply:
Oops, good point!